CMakeLists.txt

```diff
@@ -22,24 +24,71 @@ endif()
 
 include (TestBigEndian)
 TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif()
+
+configure_file(${PROJECT_SOURCE_DIR}/include/maxminddb_config.h.cmake.in 
+               ${PROJECT_SOURCE_DIR}/include/maxminddb_config.h)
+
+add_library(maxminddb
+  src/maxminddb.c
+  src/data-pool.c
+)
+add_library(maxminddb::maxminddb ALIAS maxminddb)
+
+set_target_properties(maxminddb PROPERTIES VERSION ${MAXMINDDB_SOVERSION})
+
+target_compile_definitions(maxminddb PUBLIC PACKAGE_VERSION="${PROJECT_VERSION}")
+
 if(NOT IS_BIG_ENDIAN)
-  add_definitions(-DMMDB_LITTLE_ENDIAN=1)
+  target_compile_definitions(maxminddb PRIVATE MMDB_LITTLE_ENDIAN=1)
 endif()
 
 if(MSVC)
-    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+  target_compile_definitions(maxminddb PRIVATE _CRT_SECURE_NO_WARNINGS)
+endif()
+
+if(WIN32)
+  target_link_libraries(maxminddb ws2_32)
 endif()
 
 set(CMAKE_SHARED_LIBRARY_PREFIX lib)
 set(CMAKE_STATIC_LIBRARY_PREFIX lib)
 
-configure_file(include/maxminddb_config.h.cmake.in include/maxminddb_config.h)
+set(MAXMINDB_INCLUDE_DIR
+  .
+  include
+)
 
-include_directories(. src include)
+target_include_directories(maxminddb PUBLIC
+  $<BUILD_INTERFACE:${MAXMINDB_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>
+)
 
-add_subdirectory(src)
-add_subdirectory(bin)
+set(MAXMINDB_HEADERS
+  include/maxminddb.h
+  include/maxminddb_config.h
+)
+set_target_properties(maxminddb PROPERTIES PUBLIC_HEADER "${MAXMINDB_HEADERS}")
 
-enable_testing()
-add_subdirectory(t)
+install(TARGETS maxminddb
+        EXPORT maxminddb
+        ARCHIVE DESTINATION lib
+        PUBLIC_HEADER DESTINATION include/
+)
+
+# This is required to work with FetchContent
+install(EXPORT maxminddb
+        FILE maxminddb-config.cmake
+        NAMESPACE maxminddb::
+        DESTINATION lib/cmake/maxminddb)
+
+if (BUILD_TESTING)
+  add_subdirectory(bin)
```

Comment:
`mmdblookup` isn't a test program. It should be built and installed regardless of `BUILD_TESTING`. It is a CLI tool for querying the databases.